test/unit/content-src/components/DiscoveryStreamComponents/TopSites.test.jsx

```diff
@@ -49,11 +61,126 @@ describe("Discovery Stream <TopSites>", () => {
       };
       wrapper = mount(
         <Provider store={store}>
-          <TopSites {...DEFAULT_PROPS} />
+          <TopSitesContainer {...DEFAULT_PROPS} />
         </Provider>
       );
       const oldTopSites = wrapper.find(OldTopSites);
       assert.equal(oldTopSites.props().title, "test");
     });
   });
+
+  describe("insertSpocContent", () => {
+    let insertSpocContent;
+    const topSiteSpoc = {
+      url: "foo",
+      sponsor: "bar",
+      image_src: "foobar",
+    };
+    const data = { spocs: [topSiteSpoc] };
+    const resultSpoc = {
+      customScreenshotURL: "foobar",
+      type: "SPOC",
+      label: "bar",
+      title: "bar",
+      url: "foo",
+    };
+    const pinnedSite = {
+      label: "pinnedSite",
+      isPinned: true,
+    };
+
+    beforeEach(() => {
+      const instance = wrapper.find(TopSites).instance();
+      insertSpocContent = instance.insertSpocContent.bind(instance);
+    });
+
+    it("Should return null if no data or no TopSites", () => {
+      assert.isNull(insertSpocContent(defaultTopSites, {}, "right"));
+      assert.isNull(insertSpocContent({}, data, "right"));
+    });
+
+    it("Should return null if an organic SPOC topsite exists", () => {
+      const topSitesWithOrganicSpoc = {
+        rows: [...defaultTopSiteRows, topSiteSpoc],
+      };
+
+      assert.isNull(insertSpocContent(topSitesWithOrganicSpoc, data, "right"));
+    });
+
+    it("Should return the next available non organic spoc if the first SPOC is organic", () => {
```

Comment:
nit: as commented above instead of non organic spoc something like "Should return next spoc if the first SPOC is an existing organic top site"